### PR TITLE
[V2 Loggers] config file

### PR DESCRIPTION
**Pull request by horheynm**
Screenshot 2024-01-18 at 10 53 12 AM

### Description

Config yaml file has five top level properties: version,  logging, system, performance and metric. 
version is always 2, since its for v2 pipeline
logging is the name of the defined logging (class LoggingFoo: here the name is LoggingFoo) which are singletons (one instance per any Name). 
Sys, Perf and Met are the three entry points to use the Logger. 
System is for default python (import logging) logging
Performance is for logging any hardware, benchmark,..etc metric-related log (thing that are already defined).
Metric is for input related logging (output value based on inputs).

If logging has any args for instantiation, one can define as 
```yaml
loggers:
  logger_id:
    name: FooLogger
    arg1: 1
    arg2: 2
```

For the three entry points,  the top level  after the root logger (system, metric, performance) is called the tag. Tags are used to filter out and log in any matching tag by regex. 
Under tags, we have three (four in metrics) properties, 
func is the function to apply to any value of interest to log. 
freq is the rate to log once a matching entry is encountered. 
uses: is a list to specify which logger to use, selected by logger_id

capture (metric only): key of the dict, property name to extract from the output of an operator, ran in logging middleware

```yaml
system: # Python/system level logging
  ".*": # reges tag to capture
  - func: identity # func name or path/to/file.py:func_name
    freq: 1 # rate to log
    uses: # which logger to use, point to the log id above
    - default 
metric:
  ".*":
  - func: max
    freq: 1
    uses:
    - list
    capture:  # The key, property to log. Ex rtn: Operator = AddOneOperator(...), if capture is set to value, it would record rtn.capture, if tag above is set to any regex that recognizes AddOneOperator (currently set to ".*")
    - ".*" # log every key, property that the dict, class has
```

In other words, for each root logger, what tags are we logging. With respect to tags, what function are we applying if any, at what rate and with respect to what loggers are we using

Root logger is a function of tags, tags are a fucntion of frequency, func and loggers. 
Root logger(tag(freq, func, loggers, capture))

### Examples
```yaml
version: 2 # v2 uses 2

loggers:
  list: # Logger ID
    name: ListLogger # class name or path/to/file.py:ClassName
    handler: None # Handler config, default set to default logging 
  default: 
    name: PythonLogger 
    handler: None
system: # Python/system level logging
  ".*": # reges tag to capture
  - func: identity # func name or path/to/file.py:func_name
    freq: 1 # rate to log
    uses: # which logger to use, point to the log id above
    - default 
performance:
  cpu:
  - func: identity
    freq: 1
    uses:
    - default
metric:
  ".*":
  - func: max
    freq: 1
    uses:
    - list
    capture:  # The key, property to log. Ex rtn: Operator = AddOneOperator(...), if capture is set to value, it would record rtn.capture, if tag above is set to any regex that recognizes AddOneOperator (currently set to ".*")
    - ".*"

```

Default config without any inputs
```yaml
version: 2
loggers:
  list:
    name: ListLogger
    handler: None
  default:
    name: PythonLogger
    handler: None
system:
  ".*":
  - func: identity
    freq: 1
    uses:
    - default
performance:
  cpu:
  - func: identity
    freq: 1
    uses:
    - default
metric:
  ".*":
  - func: max
    freq: 1
    uses:
    - list
    capture:
    - ".*"
```

TODO: Handlers

### CODE REVIEW

File: src/deepsparse/loggers_v2/config.py
Comment (horheynm) [reply src/deepsparse/loggers_v2/config.py:43]: The current path there is v2_loggers, before we merge to main, current loggers path will be moved to legacy, v2 will be renamed to loggers

File: src/deepsparse/loggers_v2/config.py
Comment (horheynm) [reply src/deepsparse/loggers_v2/config.py:68]: I thought old configs will be based on the old pipeline, aka failure of using v2 pipeline, so even if did use this, there is something upstream/downstream that fails no?

File: src/deepsparse/loggers_v2/config.py
Comment (horheynm) [reply src/deepsparse/loggers_v2/config.py:57]: Ok

File: src/deepsparse/loggers_v2/config.py
Comment (horheynm) [reply src/deepsparse/loggers_v2/config.py:72]: Ok

File: src/deepsparse/loggers_v2/config.py
Comment (bfineran) [reply src/deepsparse/loggers_v2/config.py:132]: let's get rid of it if we're not using the field

File: src/deepsparse/loggers_v2/config.py
Comment (bfineran) [inline src/deepsparse/loggers_v2/config.py:16]: delete

**Review by bfineran — COMMENTED** [review]
lgtm pending comments

**Review by bfineran — APPROVED** [review]

**Pull request merged**
